web_src/js/index.js

```diff
@@ -3679,6 +3680,35 @@ function initFilterBranchTagDropdown(selector) {
   });
 }
 
+function initDiffShowMore() {
+  $('#diff-show-more-files, #diff-show-more-files-stats').on('click', (e) => {
+    e.preventDefault();
+
+    if ($(e.target).hasClass('disabled')) {
+      return;
+    }
+    $('#diff-show-more-files, #diff-show-more-files-stats').addClass('disabled');
+
+    const url = $('#diff-show-more-files, #diff-show-more-files-stats').data('href');
+    $.ajax({
+      type: 'GET',
+      url,
+    }).done((resp) => {
+      if (!resp || resp.html === '' || resp.empty) {
+        $('#diff-show-more-files, #diff-show-more-files-stats').removeClass('disabled');
+        return;
+      }
+      $('#diff-too-many-files-stats').remove();
+      $('#diff-files').append($(resp).find('#diff-files li'));
+      $('#diff-incomplete').replaceWith($(resp).find('#diff-file-boxes').children());
+
+      initDiffShowMore();
```

Comment:
Is this recursion neccessary? How do you prevent infinite recursion?

Reply:
I think a delegate event listener would certainly a better option than this recursion.

Reply:
It's not recursion. It's reinitialization. The button is a completely new element if it exists at all. So you can't use a delegate.

Reply:
Indeed, it's not recursion because of the `click` handler. But delegate should still be possible and cleaner, something like:

````
$(document.body).on('click', 'selector', fn)
````

Reply:
Can also do it in vanilla JS (and I'd prefer it for new code), e.g. like:

````js
document.body.addEventListener('click', (e) => {
  if (!e.target.matches('selector') return;
  // do stuff
});
````

Reply:
> It's not a good idea to attach everything on document or body.

There is a small performance benefit of not evaluating `matches` on every click event, so I tend to agree delegate should be avoide where possible. I may be able to provide a refactor for this later in vanilla JS.

> Don't you guys wanna do frontend refactor?

Yes, we should extract stuff out of `index.js` into separate files, but it's a lot of work so can only happen gradually.